packages/ipfs/src/core/components/add-all/index.js

```diff
@@ -9,7 +9,7 @@ const { withTimeoutOption } = require('../../utils')
 module.exports = ({ block, gcLock, preload, pin, options: constructorOptions }) => {
   const isShardingEnabled = constructorOptions.EXPERIMENTAL && constructorOptions.EXPERIMENTAL.sharding
 
-  return withTimeoutOption(async function * add (source, options) {
+  return withTimeoutOption(async function * addAll (source, options) {
```

Comment:
I would like to see a deprecation warning (via `console.warn`) for `source`s that represents a single file. So that we could remove support for that in future releases.

Reply:
It seems weird to me that a new API would warn but the existing API would error (from the comment above) as you're going to break existing code and only inconvenience new code.

I think just make them both error, but this can happen in a follow up PR.